src/domains/transaction/components/VoteDetail/VoteDetail.tsx

```diff
@@ -60,7 +60,11 @@ export const VoteDetail = ({ transaction, walletAlias, ticker, isOpen, onClose }
 		const syncDelegates = () => {
 			setIsLoadingDelegates(true);
 
-			setDelegates(DelegateMapper.execute(senderWallet!, (transaction as Contracts.VoteData).votes()));
+			if (senderWallet) {
```

Comment:
Shouldn't there instead be a check that right after the `senderWallet` is retrieved that it isn't empty instead of setting defaults if it is empty? If the sender can't be found in the current profile then you are dealing either with corrupted local data or transactions that belong to a wallet outside of your profile, which would indicate that the SDK is fetching wrong information from the network.

Reply:
@faustbrian Hm ok so when that doesn't find the sender, should it be redirected off the page? or just get it from transaction `transaction.wallet()`?

Reply:
If the sender can't be found within the profile then there is some data corruption since a vote transaction is always sent from one of your own wallets that are imported into your wallet so I can't even think of a scenario where the wallet for the vote transaction wouldn't exist since you need to have the wallet imported to see the transaction. Is there even a real scenario where the wallet wouldn't be found?

Reply:
There is not, however `findByAddress` is nullable, so should we just assume that it will always be defined with `senderWallet!`?

Reply:
It should be checked if `senderWallet` is undefined and if it is throw an exception because then something is wrong with either the SDK or the data that is persisted.